src/mame/bandai/wswan_v.cpp

```diff
@@ -97,29 +113,25 @@ void wswan_video_device::device_start()
 	m_set_irq_cb.resolve();
 	m_snd_dma_cb.resolve();
 
-	if (m_vdp_type == VDP_TYPE_WSC)
-	{
-		m_vram.resize(WSC_VRAM_SIZE);
-		m_palette_vram = &m_vram[WSC_VRAM_PALETTE];
-	}
-	else
-	{
-		m_vram.resize(WS_VRAM_SIZE);
-		m_palette_vram = &m_vram[0];
-	}
-
-	std::fill(std::begin(m_vram), std::end(m_vram), 0);
+	m_vram = make_unique_clear<u16 []>(vram_size());
```

Comment:
Why not use a `const` member for the VRAM size supplied to the base constructor, rather than the overhead of a virtual member function?

Reply:
That's for avoid resizing unique_ptr variable.